benchmarks/benchmark_functions/vqe.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Benchmarks for VQE simulations.
+"""
+import pennylane as qml
+from .default_settings import _vqe_defaults
+
+qml.enable_tape()
+
+
+def benchmark_vqe(hyperparams={}):
+	"""
+	Performs VQE optimizations.
+
+	Args:
+		hyperparams (dict): hyperparameters to configure this benchmark
+
+			* 'Hamiltonian': Molecular Hamiltonian represented as a PennyLane Hamiltonian class
+
+			* 'ansatz': VQE ansatz
+
+			* 'params': Numpy array of trainable parameters that is fed into the ansatz.
+
+			* 'n_steps': Number of VQE steps
+
+			* 'device': Device on which the circuit is run, or valid device name
+
+			* 'interface': Name of the interface to use
+
+			* 'diff_method': Name of differentiation method
+
+			* 'optimize': argument for grouping the observables composing the Hamiltonian
+	"""
+
+	Hamiltonian, ansatz, params, n_steps, device, interface, diff_method, optimize = _vqe_defaults(hyperparams)
+
+	cost_fn = qml.ExpvalCost(ansatz, Hamiltonian, device, interface, diff_method, optimize)
```

Comment:
I wonder if we should keep `interface`, `diff_method`, and `optimize` packed up in a dictionary, and then unpack the dictionary when passing to `ExpvalCost`:

```python
qml.ExpvalCost(ansatz, Hamiltonian, device, **options_dict)
```

This seems more flexible. If `qml.ExpvalCost` ever adds more keywords or changes keyword order, then it won't break the benchmark.  Seeing parameters lists that long makes me wary about tolerance to errors.  Also would make it easier to read.

Reply:
Great suggestion!